### PR TITLE
chore: Bump golangci-lint to v1.56.2

### PR DESCRIPTION
**Pull request by codingllama**
Update golangci-lint to the latest patch.

I've taken this opportunity to (finally) pull the linter versions from build.assets/versions.mk, so we don't need to touch the actions on every update.

* https://github.com/golangci/golangci-lint/releases/tag/v1.56.2

### CODE REVIEW

File: .github/workflows/lint.yaml [hunk 1]
```diff
@@ -37,6 +34,16 @@ jobs:
         # We have to add the current directory as a safe directory or else git commands will not work as expected.
         run: git config --global --add safe.directory $( realpath . ) && git diff --exit-code -- go.mod go.sum api/go.mod api/go.sum
 
+      - name: Set linter versions
+        run: |
+          echo BUF_VERSION=$(cd build.assets; make print-buf-version) >> $GITHUB_ENV
+          echo GOLANGCI_LINT_VERSION=$(cd build.assets; make print-golangci-lint-version) >> $GITHUB_ENV
```
Comment (zmb3) [inline .github/workflows/lint.yaml:40]:
```suggestion
          echo BUF_VERSION=$(make -C build.assets print-buf-version) >> $GITHUB_ENV
          echo GOLANGCI_LINT_VERSION=$(make -C build.assets print-golangci-lint-version) >> $GITHUB_ENV
```

**Review by zmb3 — APPROVED** [review]

File: .github/workflows/lint.yaml (hunk 1 above)
Comment (codingllama) [reply .github/workflows/lint.yaml:40]: I had problems with `make` printing "changing to directory $foo" and then "changing to directory $bar" in-between the version number, that's why I switched to the `cd`. 🤷

**Comment by codingllama** [comment]
Thanks for the reviews, Zac!

File: .github/workflows/lint.yaml (hunk 1 above)
Comment (zmb3) [reply .github/workflows/lint.yaml:40]: Sounds good.

Comment (camscale) [reply .github/workflows/lint.yaml:40]: FYI  `make --no-print-directory -C <dir>` will suppress printing "changing to directory...", but I'm fine with a `cd` too,

**Review by camscale — APPROVED** [review]

**Review by tcsc — APPROVED** [review]

File: .github/workflows/lint.yaml (hunk 1 above)
Comment (codingllama) [reply .github/workflows/lint.yaml:40]: Yep, a `cd` seemed simpler than all of that. Thanks for the tips.

**Pull request merged**